drivers/adc/ad738x/iio_ad738x.c

```diff
@@ -0,0 +1,200 @@
+/***************************************************************************//**
+* @file   iio_ad738x.c
+* @brief  Implementation of AD738X IIO interface
+********************************************************************************
+* Copyright (c) 2024 Analog Devices, Inc.
+* Copyright (c) 2024 BayLibre, SAS.
+* All rights reserved.
+*
+* This software is proprietary to Analog Devices, Inc. and its licensors.
+* By using this software you agree to the terms of the associated
+* Analog Devices Software License Agreement.
+*******************************************************************************/
+#ifdef IIO_SUPPORT
+
+#include <stdio.h>
+#include <math.h>
+
+#include "ad738x.h"
+#include "iio_ad738x.h"
+#include "iio.h"
+#include "no_os_util.h"
+#include "no_os_alloc.h"
+#include "no_os_error.h"
+
+#define AD738X_CH(_idx) {\
+	.name = "voltage" #_idx, \
+	.ch_type = IIO_VOLTAGE,\
+	.ch_out = IIO_DIRECTION_INPUT,\
+	.indexed = true,\
+	.channel = _idx,\
+	.scan_index = _idx,\
+	.scan_type = &ad738x_iio_scan_type,\
+	.attributes = ad738x_iio_ch_attributes,\
+}
+
+#define CHN_STORAGE_BITS	16
+#define ADC_RESOLUTION		16
+#define ADC_REF_VOLTAGE		3300
```

Comment:
I guess this reference voltage may change from one project to another, from one hardware to another.
In this case it's probably better to initialize it from project through init param and not hardcode it into the driver (unless it's the only possible ref voltage, i didn't check the datasheet).